src/main/java/io/confluent/connect/hdfs/wal/WALFile.java

```diff
@@ -462,10 +461,8 @@ public Reader(Configuration conf, Option... opts) throws IOException {
         initialize(filename, file, start, len, conf, headerOnly != null);
       } catch (RemoteException re) {
         log.error("Failed creating a WAL Reader: " + re.getMessage());
-        if (re.getClassName().equals(WALConstants.LEASE_EXCEPTION_CLASS_NAME)) {
-          if (fs != null) {
-            fs.close();
-          }
+        if (fs != null) {
+          fs.close();
```

Comment:
We need to handle exceptions properly, as mentioned in an earlier comment. Here `fs.close()` can fail, yet we want to still throw `re`.